File: src/main/java/org/dicio/skill/chain/CaptureEverythingRecognizer.java
package org.dicio.skill.chain;

import org.dicio.skill.standard.StandardResult;

import java.util.List;

/**
 * A recognizer that always matches any input and results in a {@link StandardResult} on which
 * calling {@link StandardResult#getCapturingGroup(String)} always returns the input. The
 * specificity is low and the score is always 1.0.
 */
public class CaptureEverythingRecognizer extends InputRecognizer<StandardResult> {

    private String input;

    @Override
    public Specificity specificity() {
        return Specificity.low;
    }

    @Override
    public void setInput(final String input,
                         final List<String> inputWords,
                         final List<String> normalizedInputWords) {
        this.input = input;
    }

    @Override
    public float score() {
        return 1.0f;
    }

    @Override
    public StandardResult getResult() {
        return new StandardResult("", input, null) {
            @Override
            public String getCapturingGroup(final String name) {
                return input;
            }
        };
    }

    @Override
    public void cleanup() {
    }
}

File: src/main/java/org/dicio/skill/chain/CaptureEverythingRecognizer.kt
package org.dicio.skill.chain

import org.dicio.skill.standard.StandardResult

/**
 * A recognizer that always matches any input and results in a [StandardResult] on which
 * calling [StandardResult.getCapturingGroup] always returns the input. The
 * specificity is LOW and the score is always 1.0.
 */
class CaptureEverythingRecognizer : InputRecognizer<StandardResult>(Specificity.LOW) {
    private var input: String? = null

    override fun setInput(
        input: String,
        inputWords: List<String>,
        normalizedInputWords: List<String>
    ) {
        this.input = input
    }

    override fun score(): Float {
        return 1.0f
    }

    override val result: StandardResult
        get() = object : StandardResult("", input!!, mapOf()) {
            override fun getCapturingGroup(name: String): String = input!!
        }

    override fun cleanup() {
    }
}
